src/test/java/org/opensearch/geospatial/index/query/AbstractXYShapeQueryTestCase.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.query;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.opensearch.geospatial.GeospatialTestHelper.randomLowerCaseString;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertHitCount;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertSearchResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.common.geo.ShapeRelation;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.geometry.Circle;
+import org.opensearch.geometry.LinearRing;
+import org.opensearch.geometry.MultiPolygon;
+import org.opensearch.geometry.Point;
+import org.opensearch.geometry.Polygon;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.geospatial.GeospatialRestTestCase;
+import org.opensearch.geospatial.index.mapper.xypoint.XYPointFieldMapper;
+import org.opensearch.geospatial.index.mapper.xyshape.XYShapeFieldMapper;
+import org.opensearch.search.SearchHit;
+
+public abstract class AbstractXYShapeQueryTestCase extends GeospatialRestTestCase {
+    public void queryTestUsingNullShape(String indexName, String fieldName, String contentType) throws Exception {
+        createIndex(indexName, Settings.EMPTY, Map.of(fieldName, contentType));
+        String body = buildContentAsString(builder -> builder.field(fieldName, (String) null));
+        String docID = indexDocument(indexName, body);
+
+        final Map<String, Object> document = getDocument(docID, indexName);
+        assertTrue("failed to index document with type", document.containsKey(fieldName));
+        assertNull("failed to accept null value", document.get(fieldName));
+
+        deleteIndex(indexName);
```

Comment:
As discussed, lets make this as tests and add relevant abstract methods to execute the test from this class instead of subclass.

Reply:
updated as per our offline discussion